From 564c3758f4eafcc52018998d5fcdedf79015ff49 Mon Sep 17 00:00:00 2001
From: aleth <aleth@instantbird.org>
Date: Sat, 30 Jan 2016 20:56:38 +0100
Subject: [PATCH 03/28] Trac 17896: Support "Special Characters" input prompt
 on OS X

 * Bug 1151784 - Add Edit menu to the conversation window on OS X. r=nhnt11,florian

   Adding an edit menu also enables the emoji panel and dictation.
---
 im/content/instantbird.xul | 23 ++++++++++++++++++++++-
 im/content/menus.xul       |  2 +-
 im/content/menus.xul.inc   |  2 ++
 3 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/im/content/instantbird.xul b/im/content/instantbird.xul
index 15a3f98..d208921 100644
--- a/im/content/instantbird.xul
+++ b/im/content/instantbird.xul
@@ -48,7 +48,28 @@
   <script type="application/javascript" src="chrome://instantbird/content/nsContextMenu.js"/>
 
 #ifdef XP_MACOSX
-#include menus.xul.inc
+# As menus.xul.inc, but with an Edit menu.
+  <commandset id="maincommandset"/>
+  <keyset id="mainkeyset"/>
+  <menubar id="blistMenubar">
+    <menu id="menu_edit">
+      <menupopup id="menu_editpopup">
+        <menuitem id="menu_undo"/>
+        <menuitem id="menu_redo"/>
+        <menuseparator/>
+        <menuitem id="menu_cut"/>
+        <menuitem id="menu_copy"/>
+        <menuitem id="menu_paste"/>
+        <menuitem id="menu_delete"/>
+        <menuseparator/>
+        <menuitem id="menu_selectAll"/>
+        <menuseparator/>
+        <menuitem id="menu_find"/>
+        <menuitem id="menu_findAgain"/>
+      </menupopup>
+    </menu>
+  </menubar>
+  <popupset id="mainPopupSet"/>
 #endif
 
   <commandset id="conversationsCommands">
diff --git a/im/content/menus.xul b/im/content/menus.xul
index 0889ce8..894ef13 100644
--- a/im/content/menus.xul
+++ b/im/content/menus.xul
@@ -43,7 +43,7 @@
   </keyset>
 
   <menubar id="blistMenubar">
-    <menu label="&file.menu;" id="fileMenu" accesskey="&file.accesskey;">
+    <menu label="&file.menu;" id="fileMenu" accesskey="&file.accesskey;" insertbefore="menu_edit">
       <menupopup id="fileMenuPopup" onpopupshowing="menus.updateFileMenuitems();">
         <menuitem id="addBuddyMenuItem" label="&addContact;" command="cmd_addbuddy" key="addBuddykey" accesskey="&addContact.accesskey;"/>
         <menuitem id="newTabMenuItem" label="&newtab;" command="cmd_newtab" key="newtabkey" accesskey="&newtab.accesskey;"/>
diff --git a/im/content/menus.xul.inc b/im/content/menus.xul.inc
index 30aeb1f..14fc9e8 100644
--- a/im/content/menus.xul.inc
+++ b/im/content/menus.xul.inc
@@ -2,6 +2,8 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+# Note instantbird.xul contains a modified copy of this file that
+# should be kept in sync.
   <commandset id="maincommandset"/>
   <keyset id="mainkeyset"/>
   <menubar id="blistMenubar"/>
-- 
2.7.4

